rancid-tomatillos/src/apiCalls.test.js

```diff
@@ -0,0 +1,236 @@
+        rating: 8,
+        created_at: "2019-12-19T22:05:16.098Z",
+        updated_at: "2019-12-19T22:05:16.098Z"
+      }
+      ]};
+
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: true,
+          json: () => Promise.resolve(mockResponse)
+        });
+      });
+    });
+
+    it('should fetch with the correct arguments', () => {
+      const expected = 'https://rancid-tomatillos.herokuapp.com/api/v1/users/6/ratings'
+
+      fetchRatings(mockId);
+
+      expect(window.fetch).toHaveBeenCalledWith(expected);
+    });
+
+    it('should return an object with all of the user ratings', () => {
+      expect(fetchRatings(mockId)).resolves.toEqual(mockResponse);
+    });
+
+    it('SAD: should throw an error if response is not okay', () => {
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: false
+        })
+      });
+
+      expect(fetchRatings(mockId)).rejects.toEqual(Error('Invalid User ID'))
+    });
+
+    it('SAD: should throw an error if promise does not resolve', () => {
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.reject(Error('fetch failed'))
+      });
+
+      expect(fetchRatings(mockId)).rejects.toEqual(Error('fetch failed'));
+    });
+  });
+  describe('postUserRating', () => {
+    let mockResponse, mockOptions, mockMovieId, mockUserId, mockRating;
+    beforeEach(() => {
+      mockMovieId = 23;
+      mockUserId = 6;
+      mockRating = 9;
+      mockResponse = {rating: {user_id: mockUserId, movie_id: mockMovieId, rating: mockRating}};
+      mockOptions = {
+        method: 'POST',
+        body: JSON.stringify({
+        movie_id: parseInt(mockMovieId),
+        rating: parseInt(mockRating)
+        }),
+        headers: {
+        'Content-Type': 'application/json'
+        }
+      };
+
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: true,
+          json: () => Promise.resolve(mockResponse)
+        });
+      });
+    });
+
+    it('should fetch with the correct arguments', () => {
+      const expected = [`https://rancid-tomatillos.herokuapp.com/api/v1/users/${mockUserId}/ratings`, mockOptions];
+      postUserRating(mockOptions, mockUserId);
+      expect(window.fetch).toHaveBeenCalledWith(...expected);
+    });
+
+    it('should return a response of the newly posted rating', () => {
+      expect(postUserRating(mockOptions, mockUserId)).resolves.toEqual(mockResponse);
+    });
+
+    it('SAD: should throw an error message if response is not ok', () => {
+      window.fetch = jest.fn().mockImplementation(() => {
+        return Promise.resolve({
+          ok: false
+        })
+      });
+
+      expect(postUserRating(mockOptions, mockUserId)).rejects.toEqual(Error('Failure to POST User Rating'));
+    });
+    
+    it('SAD: should throw an error message if promise does not resolve', () => {
```

Comment:
Awesome work testing sad and happy paths on the apicalls! This looks like some robust testing!